src/components/MovieDetail.js

```diff
@@ -1,7 +1,47 @@
 import React from "react";
 
-const MovieDetail = () => {
-  return 'you bastart~'
+class MovieDetail extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      id: 1,
+      title: "Fake Movie Title",
+      poster_path: "https://image.tmdb.org/t/p/original//7G2VvG1lU8q758uOqU6z2Ds0qpA.jpg",
+      backdrop_path: "https://image.tmdb.org/t/p/original//oazPqs1z78LcIOFslbKtJLGlueo.jpg",
+      release_date: "2019-12-04",
+      overview: "Some overview that is full of buzzwords to attempt to entice you to watch this movie! Explosions! Drama! True love! Robots! A cute dog!", average_rating: 6,
+      genres: ["Drama"],
+      budget: 63000000,
+      revenue: 100853753,
+      runtime: 139,
+      tagline: "It's a movie!"
+    }
+  }
+
+  closeDetail = () => {
+    this.props.updateMovieId('')
+  }
+  
+  render(){
+    return (
+      <>
+        <button onClick={()=>this.closeDetail()}>X</button>
```

Comment:
WHAT A HUGE WIN!!!  we were calling that function before we put that anonymous function and it was just taking us in a loop!!!